<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  

  
  <title>大坑的博客</title>
  <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
  <meta name="keywords" content="MA→码">
<meta property="og:type" content="website">
<meta property="og:title" content="大坑的博客">
<meta property="og:url" content="http://yoursite.com/index.html">
<meta property="og:site_name" content="大坑的博客">
<meta property="og:locale" content="Dingxing方言">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="大坑的博客">
  
    <link rel="alternate" href="/atom.xml" title="大坑的博客" type="application/atom+xml">
  
  
    <link rel="icon" href="/favicon.png">
  
  
    <link href="//fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet" type="text/css">
  
  <link rel="stylesheet" href="/css/style.css">
</head>

<body>
  <div id="container">
    <div id="wrap">
      <header id="header">
  <div id="banner"></div>
  <div id="header-outer" class="outer">
    <div id="header-title" class="inner">
      <h1 id="logo-wrap">
        <a href="/" id="logo">大坑的博客</a>
      </h1>
      
        <h2 id="subtitle-wrap">
          <a href="/" id="subtitle">Kids， your road is：码畜→BUG→码狗→BUG→码农→BUG→搬砖工，Then you DIE！</a>
        </h2>
      
    </div>
    <div id="header-inner" class="inner">
      <nav id="main-nav">
        <a id="main-nav-toggle" class="nav-icon"></a>
        
          <a class="main-nav-link" href="/">Home</a>
        
          <a class="main-nav-link" href="/archives">Archives</a>
        
      </nav>
      <nav id="sub-nav">
        
          <a id="nav-rss-link" class="nav-icon" href="/atom.xml" title="RSS Feed"></a>
        
        <a id="nav-search-btn" class="nav-icon" title="Search"></a>
      </nav>
      <div id="search-form-wrap">
        <form action="//google.com/search" method="get" accept-charset="UTF-8" class="search-form"><input type="search" name="q" class="search-form-input" placeholder="Search"><button type="submit" class="search-form-submit">&#xF002;</button><input type="hidden" name="sitesearch" value="http://yoursite.com"></form>
      </div>
    </div>
  </div>
</header>
      <div class="outer">
        <section id="main">
  
    <article id="post-病原体微生物流程–v1版本：kraken-kaiju (1)" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2018/06/05/病原体微生物流程–v1版本：kraken-kaiju (1)/" class="article-date">
  <time datetime="2018-06-05T06:16:17.997Z" itemprop="datePublished">2018-06-05</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2018/06/05/病原体微生物流程–v1版本：kraken-kaiju (1)/">Test</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <h1 id="病原体微生物流程–v1版本：kraken-kaiju"><a href="#病原体微生物流程–v1版本：kraken-kaiju" class="headerlink" title="病原体微生物流程–v1版本：kraken-kaiju"></a>病原体微生物流程–v1版本：kraken-kaiju</h1><p>@(流程)[病原体|微生物|]</p>
<h3 id="流程介绍"><a href="#流程介绍" class="headerlink" title="流程介绍"></a>流程介绍</h3><blockquote>
<p>病原体微生物检测： 该流程分析脑脊液宏基因组样本，鉴定其中病原体，该病原体可能为：病毒、细菌、真菌、寄生虫。鉴定过程主要利用kraken 与 kaiju两个软件，本流程基于liuyan推荐的文章搭建：Oxford Screening CSF and Respiratory samples (‘OSCAR’): results of a pilot study to screen clinical samples from a diagnostic microbiology laboratory for viruses using Illumina next generation sequencing（<a href="https://link.springer.com/article/10.1186%2Fs13104-018-3234-8）" target="_blank" rel="noopener">https://link.springer.com/article/10.1186%2Fs13104-018-3234-8）</a></p>
</blockquote>
<ul>
<li><strong>数据库简介</strong> ：数据库简介；</li>
<li><strong>Step1 -QC</strong> ：去除低质量数据，剔除宿主–人类reads，并统计；</li>
<li><strong>Step2 -Classify</strong> ：利用kraken与kaiju将测序reads比对到数据库； </li>
<li><strong>其它</strong>：metaphlan2，megablast、Surpi_tm</li>
</ul>
<hr>
<h4 id="一、数据库简介"><a href="#一、数据库简介" class="headerlink" title="一、数据库简介"></a>一、数据库简介</h4><blockquote>
<p>鉴定病原体过程利用kraken与kaiju两个工具。<br><strong><em>kraken</em></strong>为DNA序列的鉴定工具（<a href="http://ccb.jhu.edu/software/kraken/MANUAL.html#introduction）其利用的数据库为微生物基因组，所有参考基因组都来自NCBI" target="_blank" rel="noopener">http://ccb.jhu.edu/software/kraken/MANUAL.html#introduction）其利用的数据库为微生物基因组，所有参考基因组都来自NCBI</a> reference 数据库，地址为：<a href="ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/，包括" target="_blank" rel="noopener">ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/，包括</a></p>
<ul>
<li><strong>Bacteria</strong>：113597种，路径：/Bioinfo/Database/Pathogen_reference/Reference_genome_fasta/Bacteria/</li>
<li><strong>Virus</strong>：7564种，路径/Bioinfo/Database/Pathogen_reference/Reference_genome_fasta/Virus</li>
<li><strong>Funji</strong> 254种，路径/Bioinfo/Database/Pathogen_reference/Reference_genome_fasta/Funji</li>
<li><strong>Archaea</strong> 707种，路径/Bioinfo/Database/Pathogen_reference/Reference_genome_fasta/Archaea</li>
<li><strong>Parasite</strong> 11种，只下载了文献中提到的寄生虫 /Bioinfo/Database/Pathogen_reference/Reference_genome_fasta/Parasite </li>
</ul>
</blockquote>
<blockquote>
<p><strong><em>Kaiju</em></strong>为蛋白质序列的鉴定工具（<a href="http://kaiju.binf.ku.dk/），该工具所用的数据库是可选择的：Kaiju" target="_blank" rel="noopener">http://kaiju.binf.ku.dk/），该工具所用的数据库是可选择的：Kaiju</a> can use either the set of available complete genomes from NCBI RefSeq or the microbial subset of the NCBI BLAST non-redundant protein database nr, optionally also including fungi and microbial eukaryotes，目前使用的是NCBI BLAST nr +euk103M protein sequences from nr: Bacteria, Archaea, Viruses, Fungi and microbial eukaryotes. </p>
<ul>
<li><strong>路径</strong> /Bioinfo/Database/Pathogen_reference/Kaiju_download_database/</li>
</ul>
</blockquote>
<p>####二、Step1 </p>
<blockquote>
<p>Step1，主要为QC过程，首先利用fastqc、trimmomatic去除低质量reads；然后利用Soapalign或bwa将reads比对到hg19基因组上，最后对质控过程进行统计，生成统计文件*.log，目前集成脚本Step_Qc_v0.py，该脚本路径：/Bioinfo/MDDRD2/PMO/jianglk/Pathogen_pipeline/Pipeline_kraken_kaiju/Step_1_Qc/Steq_Qc_v0.py，其中soap和bwa两个比对软件（将reads比对到hg19）是可选的。<br><figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br><span class="line">19</span><br><span class="line">20</span><br><span class="line">21</span><br><span class="line">22</span><br></pre></td><td class="code"><pre><span class="line">$ python3 Steq_Qc_v0.py -h</span><br><span class="line">usage: Steq_Qc_v0.py [-h] --name NAME --adapter ADAPTER --fastq1 FASTQ1</span><br><span class="line">                     --fastq2 FASTQ2 --outdir OUTDIR --statlog STATLOG</span><br><span class="line">                     --software SOFTWARE</span><br><span class="line"></span><br><span class="line">***Pathogen Analysis Step one --QC***</span><br><span class="line"></span><br><span class="line">optional arguments:</span><br><span class="line">  -h, --help            show this help message <span class="keyword">and</span> exit</span><br><span class="line">  --name NAME, -n NAME  样本名称</span><br><span class="line">  --adapter ADAPTER, -a ADAPTER</span><br><span class="line">                        adapter file</span><br><span class="line">  --fastq1 FASTQ1, -f1 FASTQ1</span><br><span class="line">                        测序数据 fastq1</span><br><span class="line">  --fastq2 FASTQ2, -f2 FASTQ2</span><br><span class="line">                        测序数据 fastq2</span><br><span class="line">  --outdir OUTDIR, -o OUTDIR</span><br><span class="line">                        Qc directory</span><br><span class="line">  --statlog STATLOG, -l STATLOG</span><br><span class="line">                        统计log文件</span><br><span class="line">  --software SOFTWARE, -s SOFTWARE</span><br><span class="line">                        使用bwa 或者 soap，此处输入bwa或者soap</span><br></pre></td></tr></table></figure></p>
</blockquote>
<ul>
<li><strong>bwa 与 soap</strong><blockquote>
<p>目前质控过程中reads 比对hg19的软件是可选的，soap 或者 bwa，两个软件比对结果差异还是挺大的。之后会利用RNA-seq流程验证确定哪个软件更好。</p>
</blockquote>
</li>
</ul>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">BWA</span><br><span class="line">Raw_data_size_fq1	770.18Mb</span><br><span class="line">Raw_data_size_fq2	770.18Mb</span><br><span class="line">Raw_data_reads_fastq1	10034692</span><br><span class="line">Raw_data_reads_fastq2	10034692</span><br><span class="line">Clean_data_reads_fastq1	3344744</span><br><span class="line">Clean_data_reads_fastq2	3344744</span><br><span class="line">Host_reads_fastq1	1065157</span><br><span class="line">Host_reads_fastq2	1062622</span><br><span class="line">Used_reads_fastq1	2279587</span><br><span class="line">Used_reads_fastq2	2282122</span><br><span class="line">Used_ratio_fastq1	0.2271705997553288</span><br><span class="line">Used_ratio_fastq2	0.2274232233535419</span><br></pre></td></tr></table></figure>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br></pre></td><td class="code"><pre><span class="line">Soap</span><br><span class="line">Raw_data_size_fq1	770.18Mb</span><br><span class="line">Raw_data_size_fq2	770.18Mb</span><br><span class="line">Raw_data_reads_fastq1	10034692</span><br><span class="line">Raw_data_reads_fastq2	10034692</span><br><span class="line">Clean_data_reads_fastq1	3344744</span><br><span class="line">Clean_data_reads_fastq2	3344744</span><br><span class="line">Host_reads_fastq1	357962</span><br><span class="line">Host_reads_fastq2	347512</span><br><span class="line">Used_reads_fastq1	2986782</span><br><span class="line">Used_reads_fastq2	2997232</span><br><span class="line">Used_ratio_fastq1	0.29764560785722</span><br><span class="line">Used_ratio_fastq2	0.29868699507668</span><br></pre></td></tr></table></figure>
<p>####二、Step2</p>
<blockquote>
<p>Step2，病原体鉴定过程，利用kraken 与 kaiju 两个工具，目前集成脚本Step_kraken_kaiju.py，该脚本路径：/Bioinfo/MDDRD2/PMO/jianglk/Pathogen_pipeline/Pipeline_kraken_kaiju/Step_2_kraken_kaiju/Step_kraken_kaiju.py，需要输入的是Step1中过滤低质量、宿主reads之后的 clean reads，目前测试结果可参照：<br>/Bioinfo/MDDRD2/PMO/jianglk/Pathogen_pipeline/Pipeline_kraken_kaiju/Step_2_kraken_kaiju/classify_report.txt</p>
</blockquote>
<blockquote>
<p>-<strong>注意</strong> 因服务器内存原因，目前kraken使用数据库只占总数据库的1/3，造成该流程鉴定结果不准确，因此目前流程没有测试意义。</p>
</blockquote>
<figure class="highlight python"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br><span class="line">17</span><br><span class="line">18</span><br></pre></td><td class="code"><pre><span class="line">$ python3 Step_kraken_kaiju.py -h</span><br><span class="line">usage: Step_kraken_kaiju.py [-h] --fastq1 FASTQ1 --fastq2 FASTQ2</span><br><span class="line">                            --kraken_database KRAKEN_DATABASE --name NAME</span><br><span class="line">                            --outdir OUTDIR</span><br><span class="line"></span><br><span class="line">***Use kraken <span class="keyword">and</span> kaiju to identify reads !***</span><br><span class="line"></span><br><span class="line">optional arguments:</span><br><span class="line">  -h, --help            show this help message <span class="keyword">and</span> exit</span><br><span class="line">  --fastq1 FASTQ1, -fq1 FASTQ1</span><br><span class="line">                        clean fastq1 reads</span><br><span class="line">  --fastq2 FASTQ2, -fq2 FASTQ2</span><br><span class="line">                        clean fastq2 reads</span><br><span class="line">  --kraken_database KRAKEN_DATABASE, -kd KRAKEN_DATABASE</span><br><span class="line">                        Kraken database</span><br><span class="line">  --name NAME, -n NAME  sample name</span><br><span class="line">  --outdir OUTDIR, -o OUTDIR</span><br><span class="line">                        runout dirctory</span><br></pre></td></tr></table></figure>
<p>####三、其它</p>
<blockquote>
<p><strong>测试样本</strong>：目前三个测试样本为脑炎患者脑脊液RNA测序样本，来自文献：Identification of Viruses in Cases of Pediatric Acute Encephalitis and Encephalopathy Using Next-Generation Sequencing</p>
</blockquote>
<blockquote>
<ul>
<li><strong>Metaphan2</strong>  一种微生物鉴定工具，应用于人类微生物组计划，用于物种鉴定，肠道、皮肤、阴道等等，医学应用未见报道。该工具的分析结果见： /Bioinfo/MDDRD2/PMO/jianglk/Pathogen_pipeline/Analysis/metaphlan2</li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><strong>megablast</strong>  略，还未分析</li>
</ul>
</blockquote>
<blockquote>
<ul>
<li><strong>Surpi_tm</strong>   略，因服务器配置不能达到分析要求，也尚未分析</li>
</ul>
</blockquote>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://yoursite.com/2018/06/05/病原体微生物流程–v1版本：kraken-kaiju (1)/" data-id="cji4da1gh0001b4inps7xjjwt" class="article-share-link">Share</a>
      
      
    </footer>
  </div>
  
</article>


  
    <article id="post-hello-world" class="article article-type-post" itemscope itemprop="blogPost">
  <div class="article-meta">
    <a href="/2018/06/05/hello-world/" class="article-date">
  <time datetime="2018-06-05T05:36:00.335Z" itemprop="datePublished">2018-06-05</time>
</a>
    
  </div>
  <div class="article-inner">
    
    
      <header class="article-header">
        
  
    <h1 itemprop="name">
      <a class="article-title" href="/2018/06/05/hello-world/">Hello World</a>
    </h1>
  

      </header>
    
    <div class="article-entry" itemprop="articleBody">
      
        <p>Welcome to <a href="https://hexo.io/" target="_blank" rel="noopener">Hexo</a>! This is your very first post. Check <a href="https://hexo.io/docs/" target="_blank" rel="noopener">documentation</a> for more info. If you get any problems when using Hexo, you can find the answer in <a href="https://hexo.io/docs/troubleshooting.html" target="_blank" rel="noopener">troubleshooting</a> or you can ask me on <a href="https://github.com/hexojs/hexo/issues" target="_blank" rel="noopener">GitHub</a>.</p>
<h2 id="Quick-Start"><a href="#Quick-Start" class="headerlink" title="Quick Start"></a>Quick Start</h2><h3 id="Create-a-new-post"><a href="#Create-a-new-post" class="headerlink" title="Create a new post"></a>Create a new post</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo new <span class="string">"My New Post"</span></span><br></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/writing.html" target="_blank" rel="noopener">Writing</a></p>
<h3 id="Run-server"><a href="#Run-server" class="headerlink" title="Run server"></a>Run server</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo server</span><br></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/server.html" target="_blank" rel="noopener">Server</a></p>
<h3 id="Generate-static-files"><a href="#Generate-static-files" class="headerlink" title="Generate static files"></a>Generate static files</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo generate</span><br></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/generating.html" target="_blank" rel="noopener">Generating</a></p>
<h3 id="Deploy-to-remote-sites"><a href="#Deploy-to-remote-sites" class="headerlink" title="Deploy to remote sites"></a>Deploy to remote sites</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hexo deploy</span><br></pre></td></tr></table></figure>
<p>More info: <a href="https://hexo.io/docs/deployment.html" target="_blank" rel="noopener">Deployment</a></p>

      
    </div>
    <footer class="article-footer">
      <a data-url="http://yoursite.com/2018/06/05/hello-world/" data-id="cji4da1ga0000b4inqi1u5y5o" class="article-share-link">Share</a>
      
      
    </footer>
  </div>
  
</article>


  


</section>
        
          <aside id="sidebar">
  
    

  
    

  
    
  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Archives</h3>
    <div class="widget">
      <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/06/">June 2018</a></li></ul>
    </div>
  </div>


  
    
  <div class="widget-wrap">
    <h3 class="widget-title">Recent Posts</h3>
    <div class="widget">
      <ul>
        
          <li>
            <a href="/2018/06/05/病原体微生物流程–v1版本：kraken-kaiju (1)/">Test</a>
          </li>
        
          <li>
            <a href="/2018/06/05/hello-world/">Hello World</a>
          </li>
        
      </ul>
    </div>
  </div>

  
</aside>
        
      </div>
      <footer id="footer">
  
  <div class="outer">
    <div id="footer-info" class="inner">
      &copy; 2018 Dakeng<br>
      Powered by <a href="http://hexo.io/" target="_blank">Hexo</a>
    </div>
  </div>
</footer>
    </div>
    <nav id="mobile-nav">
  
    <a href="/" class="mobile-nav-link">Home</a>
  
    <a href="/archives" class="mobile-nav-link">Archives</a>
  
</nav>
    

<script src="//ajax.googleapis.com/ajax/libs/jquery/2.0.3/jquery.min.js"></script>


  <link rel="stylesheet" href="/fancybox/jquery.fancybox.css">
  <script src="/fancybox/jquery.fancybox.pack.js"></script>


<script src="/js/script.js"></script>



  </div>
</body>
</html>